src/sio.ts

```diff
@@ -114,6 +131,10 @@ export class RPSIO {
   }
 
   writeUint32(offset: number, value: number) {
+    if (offset >= SPINLOCK0 && offset <= SPINLOCK31) {
+      const bitIndexMask = ~(1 << ((offset - SPINLOCK0) / 4));
+      this.spinLock &= bitIndexMask;
```

Comment:
let's add a return here (just a small optimization so we won't get into the `switch` below)